contrib/lkt/language/parser.py

```diff
@@ -422,19 +422,23 @@ def check_sem_internal(
     @langkit_property(return_type=T.SemanticResult.array, memoized=True)
     def check_correctness_post():
         """
-        Custom hook to implement legality checks for a given node that can
+        Custom hook to implement checks for a given node that can
```

Comment:
Why do we have only “checks” here, and “correctness” checks below? What’s the difference between “legality checks” and “correctness checks” anyway? An explanation somewhere would help.

Reply:
This comment is completely reworked later in the commit chain, so this is just a bleep. Will get rid of it

Reply:
Ok thanks